packages/core/src/kernelCoreTemplate.js

```diff
@@ -82,6 +90,44 @@
     Object.freeze(kernel)
     return kernel
 
+    function scuttleGlobalThis(extraAvoids = new Array()) {
+      let props = Object.getOwnPropertyNames(globalThis)
+      if (globalThis?.Window?.prototype) {
+        props = props.concat(Object.getOwnPropertyNames(Window.prototype))
+      }
+      if (globalThis?.EventTarget?.prototype) {
+        props = props.concat(Object.getOwnPropertyNames(EventTarget.prototype))
+      }
+
+      const avoids = [
+        // non configurables
+        'location', 'top', 'global', 'self', 'window', 'Infinity', 'NaN', 'document',
```

Comment:
just programmatically check for non-configurability

Reply:
Is that the only key by which we should skip things though?

Reply:
@naugtur I think @kumavis means specifically [line 104](https://github.com/LavaMoat/LavaMoat/pull/360/files#diff-cbec1016c903fe52b8d76860ebf3400e172b83a3374b8796bab149b3721d8064R104) and not the entire `avoids` array.

To do what you ask, I'd have to call `getOwnPropertyDescriptor` for each prop and in the worst case for all three `window`, `Window.prototype` and `EventTarget.prototype` (because I can't tell to which prototype the property belongs) to learn whether the prop is configurable or not.

In my approach I can avoid that by knowing upfront which props I should avoid, and that gives me the privilege to skip having to go through the process I just described above (more so given it's only 8 props).

Also, I feel like it's useful to always know which props we don't scuttle by a quick look at the code.

Reply:
ebb18d4773b7e7ca445791fa1b58fa9d7088be05